uvtools/dspec.py

```diff
@@ -212,63 +364,97 @@ def high_pass_fourier_filter(data, wgts, filter_size, real_delta, clean2d=False,
     # fft back to input space
     if clean2d:
         d_mdl = np.fft.fft2(_d_cl, axes=(0, 1))
+        d_res = np.fft.fft2(_d_res, axes=(0, 1))
     else:
         d_mdl = np.fft.fft(_d_cl)
+        d_res = np.fft.fft(_d_res)
 
-    # get residual in data space
-    d_res = (data - d_mdl) * ~np.isclose(wgts * win, 0.0)
+    if not linear:
```

Comment:
why get rid of the comment?

Reply:
This was a mistake. I've readded the comment.